mage_ai/data_preparation/models/triggers/__init__.py

```diff
@@ -66,6 +67,17 @@ def __post_init__(self):
         if any(env not in VALID_ENVS for env in self.envs):
             raise Exception(f'Please provide valid env values inside {list(VALID_ENVS)}.')
 
+    @property
+    def has_valid_schedule_interval(self) -> bool:
+        # Check if trigger has valid cron expression
+        if self.schedule_interval is not None and \
+            self.schedule_type == ScheduleType.TIME and \
+            self.schedule_interval not in [e.value for e in ScheduleInterval] and \
+                not croniter.is_valid(self.schedule_interval):
+            return False
```

Comment:
might be good to return False if `schedule_type == TIME` and `schedule_interval` is None

Reply:
I tested out adding a check for this as well, but decided not to include it, since the main issue with the invalid cron syntax in the `triggers.yaml` file was that an error would keep occurring on the server and prevent the trigger from actually being created (and would be hard to detect from the UI). A null `schedule_interval` is still able to create the pipeline_schedule without any issues, so the error popup isn't as critical to show.